<launch>
    <arg name="play_bag" default="true" />
    <arg name="bag_file" default="/media/usama/DATA/Backup/bags/pendulum_lowGrav.bag"/>

    <node name="player" pkg="rosbag" type="play" output="screen" args="-l -r 1 --clock $(arg bag_file)" if="$(arg play_bag)"/>

    <node pkg="pointcloud_concatenate" type="pointcloud_concatenate_node" name="pc_concat">
        <param name="target_frame" value="base_link" />    
        <param name="clouds" value="2" />
        <param name="hz" value="10" />
        <remap from="cloud_in1" to="/camera_1/depth/color/points" />
        <remap from="cloud_in2" to="/camera_2/depth/color/points" />
        <remap from="cloud_out" to="/fused_points" />
    </node>

    <node name="pcl_filter" pkg="sensor_filters" type="pointcloud2_filter_chain" output="screen">
        <rosparam command="load" file="$(find idmp_ros)/config/realsenseFilterChain.yaml" />
        <remap from="~input" to="/fused_points" />
        <remap from="~output" to="/fused_points_filt" />
    </node>

    <rosparam file="$(find idmp_ros)/config/params_live.yaml" />
    <node pkg="idmp_ros" type="idmp" name="idmp" output="screen"/>    
</launch>